app/containers/EmojiPicker/EmojiCategory.tsx

```diff
@@ -24,50 +33,46 @@ const renderEmoji = (emoji: IEmoji, size: number, baseUrl: string) => {
 			{shortnameToUnicode(`:${emoji}:`)}
 		</Text>
 	);
-};
+});
 
-class EmojiCategory extends React.Component<IEmojiCategory> {
-	renderItem(emoji: IEmoji) {
-		const { baseUrl, onEmojiSelected } = this.props;
-		return (
-			<TouchableOpacity
-				activeOpacity={0.7}
-				// @ts-ignore
-				key={emoji && emoji.isCustom ? emoji.content : emoji}
-				onPress={() => onEmojiSelected(emoji)}
-				testID={`reaction-picker-${emoji && emoji.isCustom ? emoji.content : emoji}`}>
-				{renderEmoji(emoji, EMOJI_SIZE, baseUrl)}
-			</TouchableOpacity>
-		);
-	}
-
-	render() {
-		const { emojis, width } = this.props;
+const EmojiCategory = ({ baseUrl, onEmojiSelected, emojis, width, tabsCount }: IEmojiCategory): React.ReactElement | null => {
+	const emojiSize = width ? Math.min(width / tabsCount, MAX_EMOJI_SIZE) : MAX_EMOJI_SIZE;
+	const numColumns = Math.trunc(width ? width / emojiSize : tabsCount);
+	const { colors } = useTheme();
 
-		if (!width) {
-			return null;
-		}
-
-		const numColumns = Math.trunc(width / EMOJI_SIZE);
-		const marginHorizontal = (width - numColumns * EMOJI_SIZE) / 2;
+	const renderItem = (emoji: IEmoji) => (
+		<Pressable
+			// @ts-ignore
+			key={emoji && emoji.isCustom ? emoji.content : emoji}
+			onPress={() => onEmojiSelected(emoji)}
+			testID={`reaction-picker-${emoji && emoji.isCustom ? emoji.content : emoji}`}
+			android_ripple={{ color: colors.bannerBackground, borderless: true, radius: emojiSize / 2 }}
+			style={({ pressed }: { pressed: boolean }) => ({
+				backgroundColor: isIOS && pressed ? colors.bannerBackground : 'transparent'
+			})}>
+			<Emoji emoji={emoji} size={emojiSize} baseUrl={baseUrl} />
+		</Pressable>
+	);
 
-		return (
-			<FlatList
-				contentContainerStyle={{ marginHorizontal }}
-				// rerender FlatList in case of width changes
-				key={`emoji-category-${width}`}
-				// @ts-ignore
-				keyExtractor={item => (item && item.isCustom && item.content) || item}
-				data={emojis}
-				extraData={this.props}
-				renderItem={({ item }) => this.renderItem(item)}
-				numColumns={numColumns}
-				initialNumToRender={45}
-				removeClippedSubviews
-				{...scrollPersistTaps}
-			/>
-		);
+	if (!width) {
+		return null;
 	}
-}
+
+	return (
+		<FlatList
+			// rerender FlatList in case of width changes
+			key={`emoji-category-${width}`}
+			// @ts-ignore
+			keyExtractor={item => (item && item.isCustom && item.content) || item}
```

Comment:
```suggestion
			keyExtractor={item => item.name}
```
This here doesn't make sense, given that supposedly there will always be an emoji here and this emoji always having a name is easier like this:

Reply:
Actually item here is not an object all the time. For non-custom emojis, item is a string and hence this logic.